src/topology/category/Profinite.lean

```diff
@@ -66,13 +69,64 @@ end Profinite
 def Profinite_to_Top : Profinite ⥤ Top := induced_functor _
 
 /-- The fully faithful embedding of `Profinite` in `CompHaus`. -/
-@[simps] def Profinite_to_CompHaus : Profinite ⥤ CompHaus :=
+@[simps] def Profinite.to_CompHaus : Profinite ⥤ CompHaus :=
 { obj := λ X, { to_Top := X.to_Top },
   map := λ _ _ f, f }
 
-instance : full Profinite_to_CompHaus := { preimage := λ _ _ f, f }
-instance : faithful Profinite_to_CompHaus := {}
+instance : full Profinite.to_CompHaus := { preimage := λ _ _ f, f }
+instance : faithful Profinite.to_CompHaus := {}
 
-@[simp] lemma Profinite_to_CompHaus_to_Top :
-  Profinite_to_CompHaus ⋙ CompHaus_to_Top = Profinite_to_Top :=
+@[simp] lemma Profinite.to_CompHaus_to_Top :
+  Profinite.to_CompHaus ⋙ CompHaus_to_Top = Profinite_to_Top :=
 rfl
+
+namespace Profinite
+local attribute [instance] connected_component_setoid
+
+/--
+(Implementation) The object part of the connected_components functor from compact Hausdorff spaces
+to Profinite spaces, given by quotienting a space by its connected components.
+See: https://stacks.math.columbia.edu/tag/0900
+-/
+def CompHaus.to_Profinite_obj (X : CompHaus) : Profinite :=
+{ to_Top := { α := connected_components X.to_Top.α },
+  is_compact := quotient.compact_space,
+  is_t2 := connected_components.t2,
+  is_totally_disconnected := connected_components.totally_disconnected_space }
+
+/--
+(Implementation) The bijection of homsets to establish the reflective adjunction of Profinite
+spaces in compact Hausdorff spaces.
+-/
+def Profinite.to_CompHaus_equivalence (X : CompHaus) (Y : Profinite) :
+  (CompHaus.to_Profinite_obj X ⟶ Y) ≃ (X ⟶ Profinite.to_CompHaus.obj Y) :=
+{ to_fun := λ f,
+  { to_fun := f.1 ∘ quotient.mk,
+    continuous_to_fun := continuous.comp f.2 (continuous_quotient_mk) },
+  inv_fun := λ g,
+    { to_fun := continuous.connected_components_lift g.2,
+      continuous_to_fun := continuous.connected_components_lift_continuous g.2},
+  left_inv := λ f, continuous_map.ext $ λ x, quotient.induction_on x $ λ a, rfl,
+  right_inv := λ f, continuous_map.ext $ λ x, rfl }
+
+/--
+The connected_components functor from compact Hausdorff spaces to profinite spaces,
+left adjoint to the inclusion functor.
+-/
+def CompHaus.to_Profinite : CompHaus ⥤ Profinite :=
+adjunction.left_adjoint_of_equiv Profinite.to_CompHaus_equivalence (λ _ _ _ _ _, rfl)
+
+/--
+The adjunction between CompHaus.to_Profinite and Profinite.to_CompHaus
+-/
+def to_Profinite_adj_to_CompHaus : CompHaus.to_Profinite ⊣ Profinite.to_CompHaus :=
```

Comment:
Not sure what the naming convention of adjunctions is, should this be in a namespace, and if so, which one?

Reply:
I put it in the Profinite namespace now, not sure if thats correct though...